docs/typescript/reference/dbos-class.md

```diff
@@ -0,0 +1,957 @@
+```
+
+Creates or updates an event named `key`, setting its value to `value`.
+The event can then be read by calling [`DBOS.getEvent`](#dbosgetevent) with the workflow's ID, from within another workflow, or elsewhere.
+Events are mutable.  Attempting to emit an event twice from a given workflow instance will update the value, but care should be taken to ensure that the value is calculated deterministically for consistency when workflows are recovered.
+For more information, see our [events API tutorial](../tutorials/workflow-communication-tutorial#events-api).
+
+#### `DBOS.getEvent`
+```typescript
+DBOS.getEvent<T>(workflowID: string, key: string, timeoutSeconds?: number): Promise<T | null>
+```
+
+Retrieves an event published by `workflowID` for a given `key` using the [events API](../tutorials/workflow-communication-tutorial#events-api).
+`getEvent()` returns a `Promise`.  `await`ing the promise will retrieve the value once the workflow has set the key, or return `null` if the wait times out.
+
+#### Reliability Guarantees
+
+All events are persisted to the database, so once an event it set, it is guaranteed to always be retrievable.
+
+## Finding and Retrieving Workflows
+Information about workflows that are either running or have already been completed can be retrieved from the workflow ID.  The workflow history can also be searched by other criteria.
+
+### `DBOS.getWorkflowStatus`
+`DBOS.getWorkflowStatus` retrieves the status of a single workflow, given its workflow ID.  If 
+
+```typescript
+DBOS.getWorkflowStatus(workflowID: string): Promise<WorkflowStatus>
+```
+
+The `WorkflowStatus` returned has the following field definition:
+```typescript
+interface WorkflowStatus {
+  readonly status: string; // The status of the workflow.  One of PENDING, SUCCESS, ERROR, RETRIES_EXCEEDED, ENQUEUED, or CANCELLED.
+  readonly workflowName: string; // The name of the workflow function.
+  readonly workflowClassName: string; // The class name holding the workflow function.
+  readonly workflowConfigName: string; // The name of the configuration, if the class needs configuration
+  readonly queueName?: string; // The name of the queue, if workflow was queued
+  readonly authenticatedUser: string; // The user who ran the workflow. Empty string if not set.
+  readonly assumedRole: string; // The role used to run this workflow.  Empty string if authorization is not required.
+  readonly authenticatedRoles: string[]; // All roles the authenticated user has, if any.
+  readonly request: HTTPRequest; // The parent request for this workflow, if any.
+}
+```
+
+### `DBOS.retrieveWorkflow`
+Similar to [`DBOS.getWorkflowStatus`](#dbosgetworkflowstatus), `DBOS.retrieveWorkflow` retrieves a single workflow by its id, but returns a [`WorkflowHandle`](../reference/workflow-handles.md).  This handle provides status, but also allows other workflow interactions.
+
+```typescript
+DBOS.retrieveWorkflow(workflowID: string)
+```
+
+### Searching for workflows
```

Comment:
I think we can remove this subtitle and move `DBOS.getWorkflows` and `DBOS.getWorkflowQueue` to the upper level.

Reply:
There are 4 functions in the section.  I am not sure if you meant pull them up 1 level or up 2 levels to the upper level.  Trying 1 level, which I think will work out.